### PR TITLE
Add r2 to Heron entry in processor-types file

### PR DESCRIPTION
**Pull request by abbycross**
Text from the hardware team:

    HeronR2:

    This is a revision of our original Heron processor. The chip has been redesigned to now include 156 qubits in a heavy-hexagonal lattice. While continuing to make use of the innovations of the original Heron systems, it also introduces a new TLS mitigation feature that controls the TLS environment of the chip, thereby improving coherence and stability across the whole chip.

### CODE REVIEW

File: docs/guides/processor-types.mdx [hunk 1]
```diff
@@ -23,6 +23,13 @@ At 133 qubits, Heron is an [Eagle](#eagle)-sized upgrade to [Egret](#egret) that
 
 * [Native gates and operations](native-gates): `cz, id, delay, measure, reset, rz, sx, x, if_else, for_loop, switch_case`
 
+<Admonition type="info" title="Revisions">
+  `r2` (July 2024) 
+
+  This is a revision of our original Heron processor. The chip has been redesigned to now include 156 qubits in a heavy-hexagonal lattice. While continuing to make use of the innovations of the original Heron systems, it also introduces a new TLS mitigation feature that controls the TLS environment of the chip, thereby improving coherence and stability across the whole chip.
```
Comment (beckykd) [inline docs/guides/processor-types.mdx:29]:
```suggestion
  This is a revision of the original Heron processor. The chip has been redesigned to now include 156 qubits in a heavy-hexagonal lattice. While continuing to make use of the innovations of the original Heron systems, it also introduces a new TLS mitigation feature that controls the TLS environment of the chip, thereby improving coherence and stability across the whole chip.
```

**Review by beckykd — APPROVED** [review]

**Comment by abbycross** [comment]
> Should this be edited to have R1 and R2 under the revisions and then the main heron text updated to 156 qubits?

We don't list out every revision (my understanding is that some just aren't necessary to list) but if you have, or know where to get, a nice r1 description, I can add it!

**Comment by dcmckayibm** [comment]
I think because the qubit number is changing the text then becomes confusing. The main blurb lists it as 133 and then the underneath gives a revision number

**Comment by dcmckayibm** [comment]
so I would add
r1: The first version of Heron with 133 qubits and no TLS mitigation control. (december 2023)

**Comment by abbycross** [comment]
> I think because the qubit number is changing the text then becomes confusing. The main blurb lists it as 133 and then the underneath gives a revision number

I did just change the qubit count number - thanks for highlighting that!

**Comment by abbycross** [comment]
Here's the new preview with r1 added -- what do you think @dcmckayibm ?

Screenshot 2024-07-03 at 12 55 21 PM

**Comment by dcmckayibm** [comment]
maybe we can leave off the "no tls mitigation" it's implied by the r2 description

File: docs/guides/processor-types.mdx [hunk 2]
```diff
@@ -17,12 +17,23 @@ Processor types are named for the general technology qualities that go into buil
...
 * [Native gates and operations](native-gates): `cz, id, delay, measure, reset, rz, sx, x, if_else, for_loop, switch_case`
 
+<Admonition type="info" title="Revisions">
+  `r2` (July 2024) 
+
+  This is a revision of the original Heron processor. The chip has been redesigned to include 156 qubits in a heavy-hexagonal lattice. While continuing to make use of the innovations of the original Heron systems, it also introduces a new TLS mitigation feature that controls the TLS environment of the chip, thereby improving coherence and stability across the whole chip.
+
+  `r1` (December 2023)
+
+  The first version of Heron with 133 qubits and no TLS mitigation control.
```
Comment (abbycross) [inline docs/guides/processor-types.mdx:33]:
```suggestion
  The first version of Heron with 133 qubits.
```

**Comment by dcmckayibm** [comment]
lgtm

**Pull request merged**